app/views/schools/register_ect_wizard/_use_previous_ect_choices.html.erb

```diff
@@ -5,39 +5,36 @@
 <p class="govuk-body">You used these programme choices the last time you registered an ECT.</p>
 
 <%= govuk_summary_list do |summary_list|
-  summary_list.with_row do |row|
-    row.with_key(text: 'Appropriate body')
-    row.with_value(text: @school.last_chosen_appropriate_body_name)
+  if @decorated_school.latest_registration_choices.appropriate_body.present?
+    summary_list.with_row do |row|
+      row.with_key(text: 'Appropriate body')
+      row.with_value(text: @decorated_school.latest_registration_choices.appropriate_body.name)
+    end
   end
 
   summary_list.with_row do |row|
     row.with_key(text: 'Training programme')
-    row.with_value(text: training_programme_name(@school.last_chosen_training_programme))
+    row.with_value(text: training_programme_name(@decorated_school.last_chosen_training_programme))
   end
 
-  if @school.provider_led_training_programme_chosen?
-    if @ect.lead_provider_has_confirmed_partnership_for_contract_period?(@school)
-      summary_list.with_row do |row|
-        row.with_key(text: 'Lead provider')
-        row.with_value(text: @ect.previous_lead_provider_name)
-      end
+  if @decorated_school.provider_led_training_programme_chosen? && @decorated_school.latest_registration_choices.lead_provider.present?
+    summary_list.with_row do |row|
+      row.with_key(text: 'Lead provider')
+      row.with_value(text: @decorated_school.latest_registration_choices.lead_provider.name)
+    end
 
+    if @decorated_school.latest_registration_choices.delivery_partner.present?
       summary_list.with_row do |row|
         row.with_key(text: 'Delivery partner')
-        row.with_value(text: @ect.previous_delivery_partner_name)
-      end
-    else
-      summary_list.with_row do |row|
-        row.with_key(text: 'Lead provider')
-        row.with_value(text: @ect.previous_eoi_lead_provider_name || @school.last_chosen_lead_provider&.name)
+        row.with_value(text: @decorated_school.latest_registration_choices.delivery_partner.name)
       end
     end
   end
 end %>
 
-<% if @school.provider_led_training_programme_chosen? && !@ect.lead_provider_has_confirmed_partnership_for_contract_period?(@school) %>
+<% if @decorated_school.provider_led_training_programme_chosen? && !@decorated_school.has_partnership_with?(lead_provider: @decorated_school.latest_registration_choices.lead_provider.name, contract_period: @ect.contract_start_date) %>
```

Comment:
Would this need the object rather than name?

```suggestion
<% if @decorated_school.provider_led_training_programme_chosen? && !@decorated_school.has_partnership_with?(lead_provider: @decorated_school.latest_registration_choices.lead_provider, contract_period: @ect.contract_start_date) %>
```

From what I can tell, `SchoolPartnerships::Search` expects an object, so passing the name wouldn't match anything.

Reply:
Ah yeah well spotted. Pushed a fix and a spec covering it!